src/pages/index.js

```diff
@@ -36,24 +59,103 @@ const profileEditFormPopup = new PopupWithForm({
 
 profileEditFormPopup.setEventListeners();
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    })
+    profileUserInfo.setAvatarInfo({
+      link: userData.avatar,
+    })
+  })
+
 profileEditButton.addEventListener("click", function () {
   const { name, description } = profileUserInfo.getUserInfo();
   profileTitleInput.value = name;
   profileDescriptionInput.value = description;
   profileEditFormPopup.open();
 });
 
+
+/*api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards);
+  })*/
+
+
 function handleProfileEditSubmit(userData) {
   const name = userData.title;
   const description = userData.description;
-  profileUserInfo.setUserInfo({ name, description });
-  profileEditFormPopup.close();
+  profileSubmitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```